src/Adapter/Supplier/CommandHandler/AbstractDeleteSupplierHandler.php

```diff
@@ -117,13 +117,15 @@ protected function removeSupplier(SupplierId $supplierId)
                 );
             }
 
-            if (false === $this->deleteSupplierAddress($supplierId)) {
-                throw new CannotDeleteSupplierAddressException(
-                    sprintf(
-                        'Unable to set deleted flag for supplier with id "%s" address',
-                        $supplierId->getValue()
-                    )
-                );
+            if (1 >= count($entity->getAssociatedShops())) {
+                if (false === $this->deleteSupplierAddress($supplierId)) {
```

Comment:
```suggestion
                if (1 >= count($entity->getAssociatedShops()) && false === $this->deleteSupplierAddress($supplierId)) {
```
Only one if is needed :sweat_smile: Sorry